### ISSUE TITLE
[BUG] Signs work order integration with AGOL is broken

### ISSUE THREAD

**Issue by johnclary** [cityofaustin/atd-data-tech#10489]
Our Knack Signs > AGOL integration is failing when trying to publish attachments and materials to AGOL. It appears we have orphaned child records without a parent work order.

Here's the error message:

```
Cannot insert the value NULL into column 'ATD_WORK_ORDER_ID
```

The views that are failing:
- `view_3126` (Materials)
- `view_3127` (Attachments)

**Comment by johnclary** [comment]
@dianamartin can you have someone take a look in Knack Signs & Markings to search for signs work order child records that are missing a parent? our integration is failing.

**Comment by dianamartin** [comment]
@johnclary did you want someone from SMB to check these or you mean someone from Apps?

**Comment by johnclary** [comment]
@dianamartin i assumed an Apps person needs to do it by searching in the backend?

**Comment by dianamartin** [comment]
@johnclary Yep, I'll can do it

**Comment by dianamartin** [comment]
- [ ] Queried `attachments` object - 13463 records 
- [ ]  Queried `inventory_transactions` object - `8 records` out of `52 records`
  - 3827
  - 3828
  - 10710
  - 10711
  - 10718
  - 32627
  - 32628
  - 32629

**Comment by johnclary** [comment]
thanks @dianamartin!

**Comment by johnclary** [comment]
the ETLs were still failing this morning. they are fixed bow.

the problem was that the crufty records were persisted in our postgres database. to delete them, i ran the ETLs locally with no `--date arg`, which deleted all data in postgres and agol. FYI @chiaberry

**Comment by johnclary** [comment]
The ETLs continued to fail. I was able to reproduce the issues locally, and there were indeed records in our postgres database which had missing work order IDs. this should be impossible, because we filter out such records in the Knack views that we pull data from. I checked a few of these records' histories in Knack, and they were not modified in any way that would explain why they ended up in our data store for their respective views.

So I once again deleted all records from postgres and AGOL, and re-ran the ETLs. they worked normally, without any of these errant records. Let's hope this is the end of it.

**Comment by johnclary** [comment]
The signs attachments and materials scripts are [still failing](https://austininnovation.slack.com/archives/C013G4RNJ01/p1666336514292159)—there are records in our Knack download that do not have a work order ID, because these records belong to non-signs work order types. The attachments and materials (aka inventory_transactions) objects are shared objects.

Here's what I did:
1. Locally, I downloaded all of the records for the attachments view from Postgres (view_3127)
2. I inspected them to verify that there were 11 records that just shouldn't be in the dataset. They have no value for field `field_2630` - `ATD_WORK_ORDER_ID`
